susemanager-utils/susemanager-sls/src/modules/uyuni_config.py

```diff
@@ -0,0 +1,1251 @@
+        """
+        Add or remove list of systems from the group
+
+        :param name: Group name
+        :param add_remove: True to add to the group, False to remove
+        :param system_ids: List of system ids to add or remove
+        :return: 1 on success, exception thrown otherwise
+        """
+        return self.client("systemgroup.addOrRemoveSystems", name, system_ids, add_remove)
+
+
+class UyuniSystems(UyuniRemoteObject):
+
+    def get_minion_id_map(self, refresh: bool = False) -> Dict[str, int]:
+        """
+        Map between minion ID and system internal ID of all system user have access to.
+        Context cache can be used, to avoid multiple call to the server.
+
+        :param refresh: Get new data from server, ignoring values in local context cache
+        :return: Map between minion ID and system ID of all system accessible by authenticated user
+        """
+        minions_token_key = "uyuni.minions_id_map_" + self.client.get_user()
+        if (not minions_token_key in __context__) or refresh:
+            __context__[minions_token_key] = self.client("system.getMinionIdMap")
+        return __context__[minions_token_key]
+
+
+class UyuniChildMasterIntegration:
+    """
+    Integration with the Salt Master which is running
+    on the same host as this current Minion.
+    """
+    DEFAULT_MASTER_CONFIG_PATH = "/etc/salt/master"
+
+    class FCkMinions(CkMinions):
+        """
+        Minion data matcher.
+        """
+
+        def _get_key_fingerprint(self, minion_id: str) -> str:
+            """
+            Get minion key fingerprint.
+
+            :param minion_id:
+            :return: fingerprint or an empty string if not found
+            """
+            keypath = os.path.join(self.opts['pki_dir'], self.acc, minion_id)
+            return salt.utils.crypt.pem_finger(path=keypath, sum_type=self.opts["hash_type"])
+
+        def _get_fingerprints(self, minion_ids: List[str]) -> Dict[str, str]:
+            """
+            Resolve all fingerprints.
+
+            :param minion_ids:
+            :return:
+            """
+            minions = {}
+            for mid in minion_ids:
+                minions[mid] = self._get_key_fingerprint(minion_id=mid)
+
+            return minions
+
+    def __init__(self):
+        self._minions = UyuniChildMasterIntegration.FCkMinions(salt.config.client_config(self._get_master_config()))
+
+    @staticmethod
+    def _get_master_config() -> str:
+        """
+        Return master config.
+        :return: path to salt master configuration file
+        """
+        cfg_path = UyuniChildMasterIntegration.DEFAULT_MASTER_CONFIG_PATH
+        for path in __pillar__.get("uyuni", {}).get("masters", {}).get("configs", [cfg_path]):
+            if os.path.exists(path):
+                cfg_path = path
+                break
+
+        return cfg_path
+
+    def list_minions(self, active: bool = False) -> List[str]:
```

Comment:
It seems like the `active` parameter is never actually used, I'd propose to drop it.